llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7109,6 +7109,47 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
     }
   }
 
+  {
+    Value *X;
+    Constant *C;
+    if (match(Op0, m_OneUse(m_Intrinsic<Intrinsic::abs>(m_Value(X),
+                                                        m_Constant(C)))) &&
+        match(Op1, m_Specific(X))) {
+      auto *NullValue = Constant::getNullValue(X->getType());
+      auto *MinVal = ConstantInt::get(
+          X->getType(),
+          APInt::getSignedMinValue(X->getType()->getScalarSizeInBits()));
+      bool IsIntMinPosion = C->isZeroValue();
+      switch (Pred) {
+      case CmpInst::ICMP_ULE:
+      case CmpInst::ICMP_SGE:
+        return replaceInstUsesWith(I, ConstantInt::getTrue(I.getType()));
+      case CmpInst::ICMP_UGT:
+      case CmpInst::ICMP_SLT:
+        return replaceInstUsesWith(I, ConstantInt::getFalse(I.getType()));
+      case CmpInst::ICMP_UGE:
+      case CmpInst::ICMP_SLE:
+      case CmpInst::ICMP_EQ: {
+        auto *SGE = Builder.CreateICmpSGE(X, NullValue);
+        return replaceInstUsesWith(
+            I, IsIntMinPosion
+                   ? Builder.CreateOr(SGE, Builder.CreateICmpEQ(X, MinVal))
```

Comment:
The transform doesn't match the alive2 proof:
```
----------------------------------------
define i1 @src_icmp_uge_abs(i4 %arg) {
#0:
  %abs = abs i4 %arg, 1
  %cmp = icmp uge i4 %abs, %arg
  ret i1 %cmp
}
=>
define i1 @tgt_icmp_uge_abs(i4 %arg) {
#0:
  %cmp = icmp sgt i4 %arg, 15
  ret i1 %cmp
}
Transformation seems to be correct!
```

Generally, `is_int_min_poison=1` provides more information. Thus we shouldn't emit more instructions than the case with `is_int_min_poison=0`.